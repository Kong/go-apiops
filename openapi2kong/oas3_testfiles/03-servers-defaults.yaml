# if the servers block has no hostname, then the default 'localhost'
# should be injected.
# OAS default for the url is "/" but that is useless for Kong, since there
# would be no host. Using 'localhost' is just slightly less useless :/
#
# scheme default: https
# hostname default: localhost
# port default: based on scheme

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: /
paths:
  /:
    get:
      operationId: OpsId
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
