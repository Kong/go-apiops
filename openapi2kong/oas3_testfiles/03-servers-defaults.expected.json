{
  "_format_version": "3.0",
  "services": [
    {
      "host": "localhost",
      "id": "e3a95664-95d7-5b10-bc3d-11c9cc6c2b4e",
      "name": "Simple_API_overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "5879685e-cf66-58a7-b7c5-4e8ef4699dbe",
          "methods": [
            "GET"
          ],
          "name": "Simple_API_overview-OpsId",
          "paths": [
            "~/$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_03-servers-defaults.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_03-servers-defaults.yaml"
      ]
    }
  ],
  "upstreams": []
}